### DISCUSSION TITLE
Issue with pagination

### DISCUSSION THREAD

**Discussion by KrishivGubba** [airbytehq/airbyte#44793] [Connector Builder]
Hi,
I'm building a source connector for the Bill API using Airbyte's low-code CDK.
I'm trying to implement pagination. This is what the payload of the request to the API looks like:
payload = {
    "data": "{"start":0,"max":999}",
    "devKey": "xx",
    "sessionId": "xx"
}
The "start" field represents the index of the first record that should be returned, and the "max" field is the maximum number of records that can be returned in the response. The highest value for this is 999.
I need to implement pagination, and I need the "start" field to be incremented by 999 to request a new page of records for every subsequent request until there are no more records left to be returned by the API.
MY ISSUE: How exactly can I increment the "start" field by 999, since it is nested within the payload? On top of this, the payload is sent to the API in form encoded format, and I'm not sure how to configure the YAML file in order to alter the nested "start" field for implementing the offset. Please help.

Here's the link to the API docs and the endpoint in question:
https://developer.bill.com/reference/ap-vendormgmt-listvendor

**Comment by natikgadzhi** [comment]
Hey! I know you've made another question that seems a duplicate of this, so closing.
